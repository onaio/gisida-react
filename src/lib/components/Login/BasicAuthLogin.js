import React, { Component } from 'react';
import Cookie from 'js-cookie';

export const isLoggedIn = function () {
    return Cookie.get('dsauth') === "true";
};


class BasicAuthLogin extends Component {
    constructor(props) {
        super(props);
        this.state = { loginError: false };
    }

    handleLogin(password) {
        const { appPassword } = this.props;
        if (appPassword.includes(password)) {
            Cookie.set('dsauth', true);
            location.reload();
        } else {
            this.setState({ loginError: true });
        }
    }

    render() {
        const { loginImageStyle } = this.props;

        return (
            <div className="login">
                <div>
                    <form className="login-form" onSubmit={(e) => { e.preventDefault(); this.handleLogin(this.password.value); }}>
                        <div className="form-group">
                            <div className="brand-login" style={loginImageStyle}></div>
                            {this.state.loginError === true ? <div className="alert alert-danger">Incorrect password.</div> : null}
                            <label htmlFor="password">Enter your Password</label>
                            <input className="form-control" type="password" ref={(input) => { this.password = input; }} autoFocus />
                        </div>
                        <button className="btn btn-default center-block" type="submit">Log In</button>
                    </form>
                </div>
            </div>);
    }
}

export default BasicAuthLogin