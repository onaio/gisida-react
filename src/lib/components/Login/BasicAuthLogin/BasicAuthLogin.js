import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Cookie from 'js-cookie';

export const isLoggedIn = function () {
    return Cookie.get('dsauth') === "true";
};

class BasicAuthLogin extends Component {
    constructor(props) {
        super(props);
        this.state = { loginError: false };
    }

    handleLogin(password) {
        const { appPassword } = this.props;
        if (appPassword.includes(password)) {
            Cookie.set('dsauth', true);
            location.reload();
        } else {
            this.setState({ loginError: true });
        }
    }

    render() {
        if (!this.props.appPassword || !this.props.appPassword.length) {
            return null;
        }

        return (
            <form className="login-form" onSubmit={(e) => { e.preventDefault(); this.handleLogin(this.password.value); }}>
                <div className="form-group">
                    {this.state.loginError === true ? <div className="alert alert-danger">Incorrect password.</div> : null}
                    <label htmlFor="password">Enter your Password</label>
                    <input className="form-control" type="password" ref={(input) => { this.password = input; }} autoFocus />
                </div>
                <button className="btn btn-default center-block" type="submit">Log In</button>
            </form>
        );
    }
}
BasicAuthLogin.defaultProps = {
    appPassword: []

}

BasicAuthLogin.PropTypes = {
    appPassword: PropTypes.arrayOf(PropTypes.string).isRequired
}

export default BasicAuthLogin