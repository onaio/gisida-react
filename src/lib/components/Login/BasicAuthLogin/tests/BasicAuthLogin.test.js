import { mount } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import BasicAuthLogin from './../BasicAuthLogin';

describe('/src/lib/components/Login/BasicAuthLogin', () => {
    const props = {
        appPassword: ['password123']
    }

    it('renders correctly', () => {
        const wrapper = mount(<BasicAuthLogin {...props} />)
        expect(wrapper.props()).toEqual(props)
        expect(toJson(wrapper)).toMatchSnapshot()
        wrapper.unmount()
    });

    it('renders null if basic auth password is missing', () => {
        const wrapper = mount(<BasicAuthLogin />)
        expect(wrapper.isEmptyRender()).toBe(true)
        wrapper.unmount()
    });

    it('handles incorrect password correctly', () => {
        const wrapper = mount(<BasicAuthLogin {...props} />)
        const passwordInput = wrapper.find('input');
        passwordInput.instance().value = 'password';
        passwordInput.simulate('change');
        wrapper.find('form.login-form').simulate('submit');
        expect(wrapper.find('div.alert-danger').text()).toEqual('Incorrect password.')
        wrapper.unmount()
    })

    it('handles correct password correctly', () => {
        const wrapper = mount(<BasicAuthLogin {...props} />)
        const passwordInput = wrapper.find('input');
        passwordInput.instance().value = 'password123';
        passwordInput.simulate('change');
        wrapper.find('form.login-form').simulate('submit');
        expect(wrapper.find('div.alert-danger').exists()).toBeFalsy()
        wrapper.unmount()
    })
})