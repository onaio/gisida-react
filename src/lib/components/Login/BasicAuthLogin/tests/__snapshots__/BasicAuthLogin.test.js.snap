// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/src/lib/components/Login/BasicAuthLogin renders correctly 1`] = `
<BasicAuthLogin
  appPassword={
    Array [
      "password123",
    ]
  }
>
  <form
    className="login-form"
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        htmlFor="password"
      >
        Enter your Password
      </label>
      <input
        autoFocus={true}
        className="form-control"
        type="password"
      />
    </div>
    <button
      className="btn btn-default center-block"
      type="submit"
    >
      Log In
    </button>
  </form>
</BasicAuthLogin>
`;
